eth/beacon/chains/base.py

```diff
@@ -0,0 +1,360 @@
+from __future__ import absolute_import
+
+from abc import (
+    ABC,
+    abstractmethod
+)
+from typing import (
+    Tuple,
+    Type,
+)
+
+import logging
+
+from eth_typing import (
+    Hash32,
+)
+from eth_utils import (
+    ValidationError,
+)
+
+from eth.db.backends.base import BaseAtomicDB
+from eth.exceptions import (
+    BlockNotFound,
+)
+from eth.validation import (
+    validate_word,
+)
+
+from eth.utils.datatypes import (
+    Configurable,
+)
+from eth.utils.hexadecimal import (
+    encode_hex,
+)
+from eth.utils.rlp import (
+    validate_imported_block_unchanged,
+)
+
+from eth.beacon.db.chain import (  # noqa: F401
+    BaseBeaconChainDB,
+    BeaconChainDB,
+)
+from eth.beacon.exceptions import (
+    SMNotFound,
+)
+from eth.beacon.state_machines.base import BaseBeaconStateMachine  # noqa: F401
+from eth.beacon.types.active_states import ActiveState
+from eth.beacon.types.blocks import BaseBeaconBlock
+from eth.beacon.types.crystallized_states import CrystallizedState
+from eth.beacon.validation import (
+    validate_slot,
+)
+
+
+class BaseBeaconChain(Configurable, ABC):
+    """
+    The base class for all BeaconChain objects
+    """
+    chaindb = None  # type: BaseBeaconChainDB
+    chaindb_class = None  # type: Type[BaseBeaconChainDB]
+    sm_configuration = None  # type: Tuple[Tuple[int, Type[BaseBeaconStateMachine]], ...]
+    chain_id = None  # type: int
+
+    #
+    # Helpers
+    #
+    @classmethod
+    @abstractmethod
+    def get_chaindb_class(cls) -> Type[BaseBeaconChainDB]:
+        pass
+
+    @classmethod
+    def get_sm_configuration(cls) -> Tuple[Tuple[int, Type['BaseBeaconStateMachine']], ...]:
```

Comment:
I think you can drop this method, likely getting dropped from the main `Chain` implementation (it was only added to be able to fetch this stuff over a multiprocessing pipe but that is no longer a viable API, see #1299 )